src/validate_data.py

```diff
@@ -0,0 +1,98 @@
+# clean_validate.py
+# author: Weilin Han
+# date: 2024-10-03
+
+import pandas as pd
+import pandera as pa
+
+# Function to build schema from the config file
+def build_schema_from_csv(data_config, expected_columns):
+    """Building schema for validation"""
+
+    # Ensure the data_config is a pandas dataframe
+    if not isinstance(data_config, pd.DataFrame):
+        raise TypeError("data_config must be a pandas dataframe.")
+    
+    # Ensure the data_config has following columns: column,type,max,min,category
+    required_columns = ['column', 'type', 'min', 'max','category', 'max_nullable']
```

Comment:
Could you use Panderas here to check the config 🫠

Reply:
You can even use base Python like you do, but just check that all those 6 columns exists and have those names, and that there is one row of data at least (doesn't make sense to run the function otherwise). To do the latter, you can use `data_config.empty` and as long as that is false you are good for that check.

To do the former, you can look to see that there in nothing different between the sets (column names and expected column names) and check the  `data_config.shape[1] == 6